ui_framework/doc_site/src/views/table/controlled_table.js

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+
+import {
+  KuiToolBar,
+  KuiToolBarSearchBox,
+  KuiToolBarSection,
+  KuiButton,
+  KuiButtonIcon,
+  KuiTable,
+  KuiToolBarText,
+  KuiControlledTable,
+  KuiTableRowCellLiner,
+  KuiPager,
+  KuiTableRow,
+  KuiTableRowCell,
+  KuiTableRowCheckBoxCell,
+  KuiTableHeaderCell,
+  KuiToolBarFooterSection,
+  KuiToolBarFooter,
+  KuiTableHeaderCheckBoxCell,
+  KuiTableBody,
+  KuiTableHeader,
+} from '../../../../components';
+
+import {
+  LEFT_ALIGNMENT,
+  RIGHT_ALIGNMENT
+} from '../../../../services';
+
+// Arbitrarily use the 3rd index in the sample data as the unique identifier for handling checkbox toggling.
+const CELL_ID_INDEX = 3;
+
+export class ControlledTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedItems: []
```

Comment:
I copied what we are currently doing in the landing pages. Switching it doesn't feel like a clear win to me because it complicates other functions (e.g. areAllItemsChecked would now require looping through every item in the map).  What will we want people to do in the real world, like you mention, when we have an id? It sounds like in that case we'd continue doing what we are doing with landing pages and how it is in here now?

I agree on the TableSandbox thing, though ideally I think we would have that logic in a single component so it's tested and re-usable, though I think we differ on our opinions regarding that.  If we had a component like that it could be used in this example, and the code here minimized.

So I guess I feel like since this example does not match what we'd do in the real world anywa, because it uses the nodes, not "items", then it's not really worth the time to switch it from one strawman implementation to another.  I say we leave it as is and work on a component that takes an array of objects, and a column definition, and builds a controlled table from them that is sortable, checkable, pageable, etc.  Though the details of that is for another PR and a much longer discussion! :)

thoughts?  If you feel really strongly I can switch it out bc to me it's 6 one half a dozen the other, just feels a bit pointless.